### PR TITLE
Load woocommerce-subscriptions-core 1.3.0

### LINKED ISSUE #3538: Update `subscriptions-core` to ship recent fixes in WCPay 3.5
### Describe the bug
There are various fixes and enhancements in subscriptions-core that we should get wrapped up and shipped in WCPay 3.5.

The goal of this issue is to:

- Wrap up and merge PRs in subscriptions core as needed – i.e. merge PRs that are approved, or low-risk and close.
- Tag a new version of subscriptions core.
- Include that new version in WCPay via composer update.
- Smoke testing as necessary to confirm issues are fixed and no regressions/interactions.

#### Fixes included in this update
@shendy-a8c - can you add links to all the issues (or PRs) that are fixed by subscriptions-core update? This will help with QA/testing (visibility of what's changed) and also help us track down exactly when some of these issues were fixed (when core was updated).

##### subscriptions-core 1.2.0

- https://github.com/Automattic/woocommerce-subscriptions-core/pull/46
- https://github.com/Automattic/woocommerce-subscriptions-core/pull/49

##### subscriptions-core 1.3.0

- https://github.com/Automattic/woocommerce-subscriptions-core/pull/53
- https://github.com/Automattic/woocommerce-payments/issues/3234 (https://github.com/Automattic/woocommerce-subscriptions-core/pull/55)
- https://github.com/Automattic/woocommerce-payments/issues/3412 (https://github.com/Automattic/woocommerce-subscriptions-core/pull/65)
- https://github.com/Automattic/woocommerce-payments/issues/3402 (https://github.com/Automattic/woocommerce-subscriptions-core/pull/70)
- https://github.com/Automattic/woocommerce-payments/issues/3309 (https://github.com/Automattic/woocommerce-subscriptions-core/pull/56)
- https://github.com/Automattic/woocommerce-subscriptions-core/pull/54

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes #3538
Blocked by https://github.com/Automattic/woocommerce-subscriptions-core/pull/78

#### Changes proposed in this Pull Request

Load newer woocommerce-subscriptions-core's version, `1.3.0`.
Add changes of woocommerce-subscriptions-core `1.2.0` and `1.3.0` to the changelog.

#### Testing instructions

* Checkout this branch.
* [Husky hook should run `composer install`](https://github.com/Automattic/woocommerce-payments/blob/develop/.husky/post-checkout#L8) and your `vendor/woocommerce/subscriptions-core` should be updated to be `1.3.0`. Confirm that [latest changes from `1.3.0`](https://github.com/Automattic/woocommerce-subscriptions-core/commits/1.3.0) is present, for example, [vendor/woocommerce/subscriptions-core/changelog.txt is up-to-date](https://github.com/Automattic/woocommerce-subscriptions-core/commit/10555008c58cd08b3e3bd353b04de8dc55de475c#diff-b40cd67182487ef24807d9c9268329d35fbd96aa2b0a9cae69e2e0d746b1c666R3).

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by shendy-a8c** [comment]
Thank you, @aprea for bringing it up to my attention that `composer.json` and `composer.lock` update was missing from this PR. I forgot to commit and push it. I just did in c7bf7ea.

**Review by aprea — APPROVED** [review]
This LGTM, I performed some testing on a fresh JN site with the attached testing package and everything seems to be working as expected ✅ 

[woocommerce-payments.zip](https://github.com/Automattic/woocommerce-payments/files/7758990/woocommerce-payments.zip)

File: changelog.txt [hunk 1]
```diff
@@ -6,6 +6,13 @@
 * Update - Remove task from the overview list for setting up multiple currencies.
 * Update - Return to task "Set up payments" after finishing KYC from WC-Admin.
 * Fix - Explicit currency formatting in customer-facing emails.
+* Fix: Update tooltip wording when deleting product variation.
+* Fix: Remove references to WooCommerce Subscriptions extension in the tooltips found on the Payment Methods settings table.
+* Fix: Update the Automatic Recurring Payments column on the Payment Methods table to only show which payment methods are supported by WooCommerce Subscriptions Core.
+* Fix: Prevent deprecation warnings when purchasing subscriptions products using WooCommerce Blocks.
+* Tweak: Update recurring payments copy on payment gateways page.
+* Fix: Incorrect text when filtering subscriptions to no results.
+* Changed: Subscription products must have a recurring amount greater than $0.
```
Comment (aprea) [inline changelog.txt:15]: Nitpick: Maybe change colons `:` to dashes `-` to match the existing changelog entries.

File: readme.txt [hunk 2]
```diff
@@ -104,6 +104,13 @@ Please note that our support for the checkout block is still experimental and th
 * Update - Remove task from the overview list for setting up multiple currencies
 * Update - Return to task "Set up payments" after finishing KYC from WC-Admin.
 * Fix - Explicit currency formatting in customer-facing emails.
+* Fix: Update tooltip wording when deleting product variation.
+* Fix: Remove references to WooCommerce Subscriptions extension in the tooltips found on the Payment Methods settings table.
+* Fix: Update the Automatic Recurring Payments column on the Payment Methods table to only show which payment methods are supported by WooCommerce Subscriptions Core.
+* Fix: Prevent deprecation warnings when purchasing subscriptions products using WooCommerce Blocks.
+* Tweak: Update recurring payments copy on payment gateways page.
+* Fix: Incorrect text when filtering subscriptions to no results.
+* Changed: Subscription products must have a recurring amount greater than $0.
```
Comment (aprea) [inline readme.txt:113]: Same as above, maybe change colons `:` to dashes `-` to match the existing readme entries.

File: changelog.txt (hunk 1 above)
Comment (shendy-a8c) [reply changelog.txt]: Thank you for catching that, @aprea. I think that's necessary change for consistency, not a nit =). I blindly just copy-pasted and changed the copy a bit without noticing the format.

Comment (shendy-a8c) [reply changelog.txt]: Done in 7e6d109.

File: readme.txt (hunk 2 above)
Comment (shendy-a8c) [reply readme.txt]: Done in 7e6d109.

**Pull request merged**